scico/linop/abel.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2022 by SCICO Developers
+# All rights reserved. BSD 3-clause License.
+# This file is part of the SCICO package. Details of the copyright and
+# user license can be found in the 'LICENSE' file distributed with the
+# package.
+
+"""Abel transform LinearOperator wrapping the pyabel package.
+
+Abel transform LinearOperator wrapping the
+`pyabel <https://github.com/PyAbel/PyAbel>`_ package.
+"""
+
+import math
+
+import numpy as np
+
+import jax
+import jax.experimental.host_callback
+import jax.numpy as jnp
+import jax.numpy.fft as jnfft
+
+import abel
+
+from scico.linop import LinearOperator
+from scico.typing import Array, JaxArray
+from scipy.linalg import solve_triangular
+
+
+class AbelProjector(LinearOperator):
+    r"""Abel transform projector based on `PyAbel <https://github.com/PyAbel/PyAbel>`_.
+
+    Perform Abel transform (parallel beam tomographic projection of
+    cylindrically symmetric objects) for a 2D image. The input 2D image
+    is assumed to be centered and left-right symmetric.
+    """
+
+    def __init__(self, img_shape):
+
+        self.proj_mat_quad = _pyabel_daun_get_proj_matrix(img_shape)
+
+        super().__init__(
+            input_shape=img_shape,
+            output_shape=img_shape,
+            input_dtype=np.float32,
+            output_dtype=np.float32,
+            adj_fn=self._adj,
+            jit=True,
+        )
+
+    def _eval(self, x):
+        return self._proj(x, self.proj_mat_quad).astype(self.output_dtype)
+
+    def _adj(self, x):
+        return self._bproj(x, self.proj_mat_quad).astype(self.output_dtype)
+
+    @staticmethod
+    def _proj(x: JaxArray, proj_mat_quad: Array) -> JaxArray:
+        return _pyabel_transform(x, direction="forward", proj_mat_quad=proj_mat_quad)
+
+    @staticmethod
+    def _bproj(y: JaxArray, proj_mat_quad: Array) -> JaxArray:
+        return _pyabel_transform(y, direction="transpose", proj_mat_quad=proj_mat_quad)
+
+    def inverse(self, y):
+        """Performs inverse abel transform
+
+        Args:
+            y ([JaxArray]): Input image (assumed to be a result of an Abel transform)
+
+        Returns:
+            [JaxArray]: Output of inverse Abel transform
+        """
+        return _pyabel_transform(y, direction="inverse", proj_mat_quad=self.proj_mat_quad)
+
+
+def _pyabel_transform(x, direction, proj_mat_quad, symmetry_axis=[None]):
+
```

Comment:
Is any of this code copied from `pyabel`? If so, it would be good to add an acknowledgment in a comment.

Reply:
Added a comment. Also added a citation like svmbir based on their citation instructions in https://pyabel.readthedocs.io/en/latest/readme_link.html#citation